mock.go

```diff
@@ -137,6 +138,19 @@ func (m *Mock) invoke(call *Call) {
 
 	invocation.Expectation = expected
 
+	if expected.stdin != nil {
+		// read all of stdin
+		buf, err := ioutil.ReadAll(call.Stdin)
+		if err != nil {
+			panic("error reading stdin")
+		}
+		// copy to Expectation
+		expected.readStdin = make([]byte, len(buf))
+		copy(expected.readStdin, buf)
+		// restore original stdin
+		call.Stdin = ioutil.NopCloser(bytes.NewReader(buf))
```

Comment:
I believe this works, and that `TestCallingMockWithStdinExpectedAndPassthrough` tests it.